cpp/merkletree/compact_merkle_tree.cc

```diff
@@ -21,7 +21,7 @@ CompactMerkleTree::CompactMerkleTree(SerialHasher* hasher)
 
 CompactMerkleTree::CompactMerkleTree(MerkleTree& model, SerialHasher* hasher)
     : MerkleTreeInterface(),
-      tree_(std::max(0L, int64_t(model.LevelCount()) - 1)),
+      tree_(std::max(int64_t(0), int64_t(model.LevelCount()) - 1)),
```

Comment:
I think you could avoid this (and the C-style rando-casts) by doing this?

    std::max<int64_t>(0, model.LevelCount() - 1)